sql/validation.go

```diff
@@ -20,38 +20,82 @@ import (
 	"github.com/pkg/errors"
 )
 
+type trainingDataset struct {
```

Comment:
maybe use the name `trainAndValDataset` ?

Reply:
Also, this source file name `validation.go` seems is not what this file does.

Reply:
> Also, this source file name `validation.go` seems is not what this file does.

Good point. How about `split.go` or `create_train_val.go`?